datalad/cli/helpers.py

```diff
@@ -255,17 +255,23 @@ def _parse_overrides_from_cmdline(cmdlineargs):
     # errors: we need a section, a variable, and a value at minimum
     # otherwise we break our own config parsing helpers
     # https://github.com/datalad/datalad/issues/3451
-    noassign_expr = re.compile(r'[^\s]+\.[^\s]+=[\S]+')
+    assign_expr = re.compile(r'[^\s]+\.[^\s]+=[\S]+')
+    unset_expr = re.compile(r':[^\s]+\.[^\s=]+')
     noassign = [
         o
         for o in cmdlineargs.cfg_overrides
-        if not noassign_expr.match(o)
+        if not (assign_expr.match(o) or unset_expr.match(o))
     ]
     if noassign:
         lgr.error(
             "Configuration override without section/variable "
-            "or value assignment (must be 'section.variable=value'): %s",
+            "or unset marker or value assignment "
+            "(must be '[:]section.variable[=value]'): %s",
```

Comment:
```suggestion
            "(must be '(:section.variable|section.variable=value)'): %s",
```
if agree that it is clearer